### PR TITLE
Use pip

### LINKED ISSUE #14: pytz-2017.3-py_1 does not install the right egg file for py27
I am trying to build my custom package with conda-build. It fails when setuptools tries to download pytz as dependency:
```
$conda-build --python 2.7 --output-folder conda recipe
Attempting to finalize metadata for mspy
INFO:conda_build.metadata:Attempting to finalize metadata for mspy
BUILD START: ['mspy-0.8-py27hfbc075d_0.tar.bz2']

The following NEW packages will be INSTALLED:

    blas:            1.1-openblas                  conda-forge
    boost:           1.64.0-py27_4                 conda-forge
    boost-cpp:       1.64.0-1                      conda-forge
    bzip2:           1.0.6-1                       conda-forge
    ca-certificates: 2017.7.27.1-0                 conda-forge
    cairo:           1.14.6-5                      conda-forge
    certifi:         2017.7.27.1-py27_0            conda-forge
    curl:            7.55.1-0                      conda-forge
    expat:           2.1.0-3                       conda-forge
    fontconfig:      2.12.1-5                      conda-forge
    freetype:        2.7-2                         conda-forge
    freexl:          1.0.2-2                       conda-forge
    gdal:            2.2.2-py27_0                  conda-forge
    geos:            3.6.2-0                       conda-forge
    gettext:         0.19.7-1                      conda-forge
    giflib:          5.1.4-0                       conda-forge
    glib:            2.51.4-0                      conda-forge
    hdf4:            4.2.13-0                      conda-forge
    hdf5:            1.8.18-2                      conda-forge
    icu:             58.1-1                        conda-forge
    jpeg:            9b-2                          conda-forge
    json-c:          0.12.1-0                      conda-forge
    kealib:          1.4.7-3                       conda-forge
    krb5:            1.14.2-0                      conda-forge
    libdap4:         3.18.3-2                      conda-forge
    libffi:          3.2.1-3                       conda-forge
    libgdal:         2.2.2-5                       conda-forge
    libgfortran:     3.0.0-1                                  
    libiconv:        1.14-4                        conda-forge
    libkml:          1.3.0-2                       conda-forge
    libnetcdf:       4.4.1.1-9                     conda-forge
    libpng:          1.6.28-2                      conda-forge
    libpq:           9.6.3-0                       conda-forge
    libspatialite:   4.3.0a-17                     conda-forge
    libssh2:         1.8.0-2                       conda-forge
    libtiff:         4.0.7-1                       conda-forge
    libxml2:         2.9.5-1                       conda-forge
    ncurses:         5.9-10                        conda-forge
    numpy:           1.11.3-py27_blas_openblas_202 conda-forge [blas_openblas]
    olefile:         0.44-py27_0                   conda-forge
    openblas:        0.2.19-2                      conda-forge
    openjpeg:        2.1.2-3                       conda-forge
    openssl:         1.0.2l-0                      conda-forge
    pandas:          0.21.0-py27_0                 conda-forge
    pcre:            8.39-0                        conda-forge
    pillow:          4.3.0-py27_1                  conda-forge
    pip:             9.0.1-py27_0                  conda-forge
    pixman:          0.34.0-0                      conda-forge
    poppler:         0.52.0-3                      conda-forge
    poppler-data:    0.4.7-0                       conda-forge
    proj4:           4.9.3-4                       conda-forge
    python:          2.7.14-1                      conda-forge
    python-dateutil: 2.6.1-py27_0                  conda-forge
    pytz:            2017.3-py_1                   conda-forge
    readline:        6.2-0                         conda-forge
    scipy:           0.19.1-py27_blas_openblas_202 conda-forge [blas_openblas]
    setuptools:      36.6.0-py27_1                 conda-forge
    six:             1.11.0-py27_1                 conda-forge
    sqlite:          3.13.0-1                      conda-forge
    tk:              8.5.19-2                      conda-forge
    util-linux:      2.21-0                                   
    wheel:           0.30.0-py_1                   conda-forge
    xerces-c:        3.1.4-3                       conda-forge
    xz:              5.2.3-0                       conda-forge
    zlib:            1.2.11-0                      conda-forge
<snip>
RuntimeError: Setuptools downloading is disabled in conda build. Be sure to add all dependencies in the meta.yaml  url=https://pypi.python.org/simple/pytz/
```

When I create an environment using the same packages as the packages that conda-build is trying to use above, the resulting environment does not have an egg for python2.7:
```
$find /path/to/envs/msp_try/ -name \*pytz\*
/path/to/envs/msp_try/conda-meta/pytz-2017.3-py_1.json
/path/to/envs/msp_try/lib/python2.7/site-packages/pytz
/path/to/envs/msp_try/lib/python2.7/site-packages/pytz-2017.3-py3.6.egg-info
```

If I go ahead and install the arch package, I see the right egg file:
```
conda install -m msp_try pytz-2017.3-py27_0.tar.gz
$find /path/to/envs/msp_try/ -name \*pytz\*
/path/to/envs/msp_try/conda-meta/pytz-2017.3-py27_0.json
/path/to/envs/msp_try/lib/python2.7/site-packages/pytz
/path/to/envs/msp_try/lib/python2.7/site-packages/pytz-2017.3-py2.7.egg-info
```

If I call ```python setup.py install``` in this custom built environment, setuptools does not try to download the pytz package from pypi.

My custom package requires pytz through pandas. pandas is a direct requirement of my custom package.

### PR DESCRIPTION
**Pull request by ocefpaf**
Hopefully this will close #14. Ping @microe.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -9,14 +9,15 @@ source:
   sha256: fae4cffc040921b8a2d60c6cf0b5d662c1190fe54d718271db4eb17d44a185b7
 
 build:
-  number: 1
+  number: 2
   noarch: python
-  script: python setup.py install --single-version-externally-managed --record=record.txt
+  script: python -m pip install --no-deps --ignore-installed .
```
Comment (microe) [inline recipe/meta.yaml:14]: :+1: I like the --ignore-installed. I'll give this a shot. Thanks for sticking with this one.

Comment (ocefpaf) [reply recipe/meta.yaml:14]: If this works we (conda-forge) should start migrating to use it instead of `python setup.py install`.

Comment (microe) [reply recipe/meta.yaml:14]: I have built a pytz package with this recipe and can confirm that my custom package can be built and then installed and run with this pytz package. I can confirm that the pytz package does not install an egg, instead installing a wheel. As far as I am concerned issue #14 is resolved.

Comment (ocefpaf) [reply recipe/meta.yaml:14]: Thanks for the feedback @microe! For some odd reason CircleCi is taking too long but we'll merge this ASAP.

**Review by microe — APPROVED** [review]

**Pull request merged**